### PR TITLE
 Feature/46/Add Components to Dashboard Page

### PR DESCRIPTION
**Pull request by eden-chan**

### CODE REVIEW

**Review by LemonFace0309 — CHANGES_REQUESTED** [review]
I noticed you have merge conflicts. This is probably because you branched off of your old branch instead of main. You might want to run `git rebase --onto main <old_branch_name_whatever_it_was>` and fix whatever merge conflcits you have locally before your next push to repo. Other than that, ping me once this branch is ready to be merged and not just a draft.

**Comment by LemonFace0309** [comment]
@eden-chan you can continue pushing changes into this branch before I approve, instead of creating a new branch that has the commit history of this one.

**Comment by LemonFace0309** [comment]
@eden-chan please make sure you resolve merge conflicts asap as well, instead of leaving it for the reviewer. 
image

In this case, I think you have this because you either didn't branch off of main, or when you did, your main branch was a few commits behind the remote one. I will resolve it this time but please keep this in mind.

File: components/Shared/Card.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+      <Paper className="rounded-xl">
+
+       {children}
+      
+        
+    </Paper>)
+      
+      
+
```
Comment (LemonFace0309) [inline components/Shared/Card.js:18]: why so spaced out?

**Comment by LemonFace0309** [comment]
@eden-chan after quickly scanning through your branch, it looks like you still have remnants of your tailwind css experiments. Please clean this up and use eslint to make sure code is consistent.

**Review by LemonFace0309 — CHANGES_REQUESTED** [review]

**Comment by LemonFace0309** [comment]
Also, when doing `npm i` we get this error: `npm WARN deprecated @reactchartjs/react-chart.js@1.0.0-rc.4: This package is deprecated. Please switch to 'react-chartjs-2'`

@eden-chan do you think you can quickly update the package and adjust imports to fix this error?

File: components/Dashboard/DashboardContainer.js [hunk 2]
```diff
@@ -1,36 +1,131 @@
...
+              </Grid>
+              <Grid item xs={12}>
+                <DashboardCard
+                  variant="dark"
+                  spaceName="Create a Space"
+                  description="insert some sort of tagline or feature description "
+                />
+              </Grid>
+            </Grid>
+            <Grid item xs={6} container direction="column" spacing={3}>
```
Comment (jimmyygit) [inline components/Dashboard/DashboardContainer.js:73]: I think this is the culprit for poor responsiveness. We are using `Grid` but were not taking full advantage of the flexibility it offers. Particularly with the "column width" props ie `xs` , `sm`, `md`. 
Web grid columns contain 12 fractions. We usually want adjacent grids to sum to 12. Common design pattern would be `xs={12} md={6}` which basically means "this grid col should be full width on x-small to medium and half width on medium and up"

image

File: pages/dashboard/index.js [hunk 3]
```diff
@@ -1,16 +1,50 @@
...
+        <Sidebar />
       </Grid>
-      <Grid item xs={3}>
-        <DashboardContainer/>
+      <Grid item xs={9} container direction="column">
+        <Grid item>
+          <DashboardContainer />
+        </Grid>
+        <Grid item container>
+          <Grid item>
```
Comment (jimmyygit) [inline pages/dashboard/index.js:23]: I think the nesting here is a bit redundant. One grid should do :)

File: pages/spaces/study-space-demo.js [hunk 4]
```diff
@@ -36,86 +35,75 @@ const theme = createMuiTheme({
...
+            className="flex flex-row space-x-5 ">
+            <VideocamOffIcon />
+            <MicIcon />
+          </Box>
+        </Grid>
+
+        {/* MUI grid can be an item AND a container. */}
+        <Grid className="col-span-8">
+          <Container className="mb-24">
+            <Grid className="grid grid-cols-6 ml-5 space-x-5 space-y-5 ">
```
Comment (jimmyygit) [inline pages/spaces/study-space-demo.js:87]: I think for the sake of consistency, we should define column widths in the props instead of classNames

**Review by jimmyygit — COMMENTED** [review]
rock solid image

addressing the responsive part, I think just adding the grid props should do
https://material-ui.com/components/grid/#basic-grid

sorry if i commented on parts u didnt write, I dont know if the commits were mumbo jumbo so I just looked at everything

**Review by LemonFace0309 — APPROVED** [review]

**Pull request merged**